crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -750,6 +782,7 @@ impl<'db, 'ast> SemanticIndexBuilder<'db, 'ast> {
     /// Records that all remaining statements in the current block are unreachable.
     fn mark_unreachable(&mut self) {
         self.current_use_def_map_mut().mark_unreachable();
+        self.check_lazy_snapshots(self.current_scope());
```

Comment:
We compare terminal (!= end-of-scope) bindings with snapshotted (captured) bindings to determine the completeness of the snapshot. We need to call `check_lazy_snapshots` even when the scope becomes unreachable.